studiopress/genesis-custom-blocks
# Add InnerBlocks feature

Fixes [GF-3072](https://wpengine.atlassian.net/browse/GF-3072)

#### Changes
* Uses Gutenberg's [InnerBlocks](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/)

<img width="851" alt="here-iis-the-inner-blocks-field" src="https://user-images.githubusercontent.com/4063887/118318482-12143700-b4bf-11eb-89a5-f600c6af3a1d.png">

<img width="720" alt="the-inner-blocks" src="https://user-images.githubusercontent.com/4063887/118318630-425bd580-b4bf-11eb-9fbc-b312d125a0d8.png">

#### Testing instructions
1. `composer install && npm install && npm run build`
2. `/wp-admin` > Custom Blocks > Add New
3. Add a field with the 'Field Type' of 'Inner Blocks': 
![field-with-inner-blocks](https://user-images.githubusercontent.com/4063887/123907261-417de680-d93b-11eb-92cf-17ff3710b9fd.png)
4. Add another field with some other 'Field Type,' like 'Image'
5. In the Template editor, put the field names in double brackets, like:
![this-](https://user-images.githubusercontent.com/4063887/123907454-8bff6300-d93b-11eb-82f8-ffdb267cecf4.png)
6. Publish the block and create a new post
7. Add the block you just created: 
![ijoidajfads](https://user-images.githubusercontent.com/4063887/123907618-d97bd000-d93b-11eb-9f16-253517237498.png)
8. Click the + icon: 
![ajsodifjasdoifj](https://user-images.githubusercontent.com/4063887/123907703-f6180800-d93b-11eb-8d25-7d91199dfe75.png)
9. Add whatever block you want, maybe a quote block:  
![hisodfasd](https://user-images.githubusercontent.com/4063887/123907839-2364b600-d93c-11eb-8bc6-7fbadf6571dc.png)
10. Preview the post's front-end: 
![click-to-preview](https://user-images.githubusercontent.com/4063887/123992604-613ff980-d991-11eb-8b24-6b8eb1bc2afd.png)
11. Expected: The quote you entered displays: 
![the-quote-displays](https://user-images.githubusercontent.com/4063887/123992782-87659980-d991-11eb-9209-7680ce28e25b.png)
